pkg/porter/publish_test.go

```diff
@@ -145,51 +147,56 @@ func TestPublish_getNewImageNameFromBundleReference(t *testing.T) {
 	})
 }
 
-func TestPublish_UpdateBundleWithNewImage(t *testing.T) {
+func TestPublish_UpdateRelocationMapping(t *testing.T) {
 	p := NewTestPorter(t)
 	defer p.Close()
 
-	bun := cnab.NewBundle(bundle.Bundle{
-		Name: "mybuns",
-		InvocationImages: []bundle.InvocationImage{
-			{
-				BaseImage: bundle.BaseImage{
-					Image:  "myorg/myinvimg",
-					Digest: "abc",
-				},
-			},
-		},
-		Images: map[string]bundle.Image{
-			"myimg": {
-				BaseImage: bundle.BaseImage{
-					Image:  "myorg/myimg",
-					Digest: "abc",
-				},
-			},
-		},
-	})
+	originImg := "myorg/myinvimg"
 	tag := "myneworg/mynewbuns"
-
 	digest, err := image.NewDigest("sha256:6b5a28ccbb76f12ce771a23757880c6083234255c5ba191fca1c5db1f71c1687")
 	require.NoError(t, err, "should have successfully created a digest")
 
-	// update invocation image
-	newInvImgName, err := getNewImageNameFromBundleReference(bun.InvocationImages[0].Image, tag)
-	require.NoError(t, err, "should have successfully derived new image name from bundle tag")
+	testcases := []struct {
+		description   string
+		relocationMap relocation.ImageRelocationMap
+		layout        registry.Layout
+		hasError      bool
```

Comment:
Let's change this to use `wantError string` so that we can check that it failed for the expected reason.